lib/binding_rust/lib.rs

```diff
@@ -1933,10 +1933,23 @@ impl QueryCursor {
         self
     }
 
+    /// Set the maximum start depth for a cursor.
+    ///
+    /// This prevents cursors from exploring children nodes at a certain depth.
+    /// Note if a pattern includes many children, then they will still be checked.
+    ///
+    /// The zero max depth value can be used as a special behavior and help to _destructure_
+    /// a subtree by staying on a node and using captures for interested parts.
+    ///
+    /// Set to `None` to remove the maximum start depth.
     #[doc(alias = "ts_query_cursor_set_max_start_depth")]
-    pub fn set_max_start_depth(&mut self, max_start_depth: u32) -> &mut Self {
+    pub fn set_max_start_depth(&mut self, max_start_depth: Option<u32>) -> &mut Self {
         unsafe {
-            ffi::ts_query_cursor_set_max_start_depth(self.ptr.as_ptr(), max_start_depth);
+            if let Some(depth) = max_start_depth {
+                ffi::ts_query_cursor_set_max_start_depth(self.ptr.as_ptr(), depth);
+            } else {
+                ffi::ts_query_cursor_set_max_start_depth(self.ptr.as_ptr(), u32::MAX);
+            }
```

Comment:
You may prefer the above, but another way to write this could be:
```rust
ffi::ts_query_cursor_set_max_start_depth(self.ptr.as_ptr(), max_start_depth.unwrap_or(u32::MAX));
```

Reply:
Good point, thanks! The idea of writing functionally doesn't always cross my mind.